### PR TITLE
Improvements to the Cuda Core C library infrastructure

### PR DESCRIPTION
**Pull request by miscco**
This makes it easier to integrate into other infrastructures by using libfmt instead of std::format which is only available in gcc-13

Furthermore we add a filter for changes that are purely in the Cuda Core C library to not trigger the full CI

See #2334

### CODE REVIEW

**Comment by miscco** [comment]
This successfully detects changes to the c library:
```
### Modifications in project?

|     | Project
|-----|---------
| +/- | CCCL Infrastructure
|     | libcu++
|     | CUB
|     | Thrust
|     | CUDA Experimental
|     | pycuda
| +/- | CUDA C Core Library 
```

**Review by bernhardmgruber — APPROVED** [review]

File: ci/inspect_changes.sh [hunk 1]
```diff
@@ -37,6 +38,7 @@ declare -A dependencies=(
   [thrust]="cccl libcudacxx cub"
   [cudax]="cccl libcudacxx"
   [pycuda]="cccl libcudacxx cub thrust cudax"
```
Comment (leofang) [inline ci/inspect_changes.sh:40]: `cuda.parallel` depends on the C library:
```suggestion
  [pycuda]="cccl libcudacxx cub thrust c"
```

**Review by leofang — COMMENTED** [review]
Thanks, Michael! Left a comment based on my understanding of the component relation.

**Review by gevtushenko — CHANGES_REQUESTED** [review]
CCCL/c is producing a binary, not sources, so there's no immediate need for a wide compiler matrix support. Regarding the fmt dependency, we fetch it as part of libcu++ codegen project once and fetch it in different way as part of nvbench setup in our benchmarks. I don't know if this can create conflicts for development preset. Let's focus this PR on CI dependencies and figure out CCCL-wide fmt availablility after https://github.com/NVIDIA/cccl/pull/2338 is merged.

**Review by alliepiper — APPROVED** [review]
Signing off on the `inspect_changes.sh` update.

**Review by gevtushenko — APPROVED** [review]

**Pull request merged**